### DISCUSSION TITLE
Should @nuxt/ui-template's "license": field really be CC-BY-ND 4.0?

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#27233] [Questions] [answered]
Let's start by mentioning that it is a valid SPDX identifier: https://spdx.org/licenses/CC-BY-ND-4.0.html

but I'm using the package as software, not as content (i.e. downloading it as a dependency of Nuxt). My company's list of allowed licenses only lists software licenses, and rejects the CC-BY-ND-4.0 because it's not on the allowed list. This happens even when just downloading nuxt, since @nuxt/ui-templates is a dependency of @nuxt/schema, which is a dependency of nuxt itself.

That is obviously just my problem, but I was wondering if the package.json shouldn't include the MIT license instead, while keeping the CC-by-ND-4.0 for the design resources? That way, there'd be uniformity of package licenses (MIT) across Nuxt deps

**Comment by danielroe** [comment]
Yes, that's a valid point. However, this package is no longer separately published by Nuxt, and in v3.12 `@nuxt/ui-templates` is not going to be a dependency. Instead the templates are inlined in `nuxt` and `@nuxt/schema` and covered by the MIT licence...

**Reply by TheAlexLichter** [reply]
Could we then retrospectively/additionally change the licenses so "older" Nuxt versions are still complying to "OSS"-only licenses?

**Reply by ghost** [reply]
Because the license is in package.json physically, you'd have to unpublish the package and republish. On NPM, this means having to e-mail NPM and ask them to remove the old version.

The easiest thing to do for people that are staying on 3.11 would be to publish a new patch (x.x.1) version with the updated license. That way, the @nuxt/schema package's `"@nuxt/ui-templates": "^1.3.3"` would just download 1.3.4 with the updated license, at least.

e.g. here's a case where the package's license was updated mid-life:
https://github.com/kriszyp/json-schema/pull/78

Or the users can just wait till 3.12 release as Daniel mentioned

**Reply by TheAlexLichter** [reply]
> The easiest thing to do for people that are staying on 3.11 would be to publish a new patch (x.x.1) version with the updated license.

That was what I had in mind 😋

**Reply by danielroe** [reply, marked as answer]
Published v1.3.4 with MIT licence ✅

**Comment by ghost** [comment]
There's also another potential licensing issue in og-image (dependency of nuxt seo): https://github.com/nuxt-modules/og-image/issues/212

**Reply by TheAlexLichter** [reply]
Let's track there 👍🏻
